src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -474,6 +471,10 @@ def _generate_zenml_pipeline_dockerfile(
 
         lines.append("RUN chmod -R a+rw .")
 
+        if docker_settings.user:
+            lines.append(f"USER {docker_settings.user}")
+            lines.append(f"RUN chown -R {docker_settings.user} .")
```

Comment:
Why is this necessary? Seems like this is the meat of the PR :D

Reply:
This user might not have the necessary permissions to install the `pip` packages in the above commands.
The `chown` command makes sure that the `/app` directory (which contains the user files/code) is owned by the provided user instead of `root`